src/logics/Logic.cc

```diff
@@ -918,7 +870,7 @@ PTRef Logic::mkEq(vec<PTRef>& args) {
         return mkAnd(binaryEqualities);
     }
     assert(args.size() == 2);
-    if (isConstant(args[0]) && isConstant(args[1]))
+    if (isConstant(args[0]) && isConstant(args[1]) && (not isIte(args[0])) && (not isIte(args[1])))
```

Comment:
Again the situation of constant vs ITE. So, can it happen that ITE is a constant or not?